File: Aplikacja/Serwer/CsharpServerApp/oto-auto-c-sharp-server/oto-auto-c-sharp-server/Logic/Offers/Api/OfferController.cs
using Microsoft.AspNetCore.Mvc;
using oto_auto_c_sharp_server.Entities;
using oto_auto_c_sharp_server.Logic.Offers.Models;
using oto_auto_c_sharp_server.Logic.Offers.Models.CreateOffer;

namespace oto_auto_c_sharp_server.Logic.Offers.Api;

[ApiController]
[Route("api/offer")]
public class OfferController: ControllerBase
{
    private readonly IOfferAdapter _offerAdapter;

    public OfferController(IOfferAdapter offerAdapter)
    {
        _offerAdapter = offerAdapter;
    }

    [HttpGet()]
    public async Task<ActionResult<IEnumerable<Offer>>> GetAllOffers()
    {
        var offers = await _offerAdapter.GetAllOffers();
        return Ok(offers);
    }

    [HttpGet("with_vehicles")]
    public async Task<ActionResult<IEnumerable<Offer>>> GetOffersWithVehicles()
    {
        var offers = await _offerAdapter.GetOffersWithVehicles();
        return Ok(offers);
    }
    
    [HttpGet("with_vehicle/{offerId}")]
    public async Task<ActionResult<Offer>> GetOfferWithVehicleByOfferId(int offerId)
    {
        var offer = await _offerAdapter.GetOfferWithVehicleByOfferId(offerId);
        return Ok(offer);
    }

    [HttpGet("awarded")]
    public async Task<ActionResult<List<OfferCardComponentModel>>> GetAwardedOffersCardComponentModel()
    {
        var awardedOffers = await _offerAdapter.GetAwardedOffers();
        return Ok(awardedOffers);
    }

    [HttpGet("{offerId}")]
    public async Task<ActionResult<OfferActivityComponentModel>> GetOfferById(int offerId)
    {
        var offer = await _offerAdapter.GetOfferById(offerId);
        return Ok(offer);
    }

    [HttpPost("create")]
    public async Task<ActionResult<CreateOfferResponse>> CreateOffer(CreateOfferRequest request)
    {
        var offerId = await _offerAdapter.CreateOffer(request);
        return Ok(offerId);
    }

    [HttpGet("filtered")]
    public async Task<ActionResult<IEnumerable<OfferPreview>>> GetFilteredOffers()
    {
        var offers = await _offerAdapter.GetFilteredOffers();
        return Ok(offers);
    }
}   

File: Aplikacja/Mobile/OtoAuto/app/src/main/java/org/masters_thesis/otoauto/logic/offer/api/OfferController.kt
package org.masters_thesis.otoauto.logic.offer.api

import org.masters_thesis.otoauto.model.OfferActivityComponentModel
import org.masters_thesis.otoauto.model.OfferCardComponentModel
import org.masters_thesis.otoauto.model.OfferModel
import org.masters_thesis.otoauto.model.createOffer.CreateOfferRequest
import org.masters_thesis.otoauto.model.createOffer.CreateOfferResponse
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface OfferController {

    @GET("/api/offer/with_vehicles")
    fun getOffersWithVehicles(): Call<List<OfferModel>>

    @GET("/api/offer/with_vehicle/{offerId}")
    fun getOfferWithVehicleByOfferId(@Path("offerId") offerId: Int): Call<OfferModel?>

    @GET("/api/offer/awarded")
    fun getAwardedOffers(): Call<List<OfferCardComponentModel?>>

    @GET("/api/offer/{offerId}")
    fun getOfferById(@Path("offerId") offerId: Int): Call<OfferActivityComponentModel?>

    @POST("/api/offer/create")
    fun createOffer(@Body createOfferRequest: CreateOfferRequest): Call<CreateOfferResponse>
}